spec/src/main/asciidoc/mpconfig.asciidoc

```diff
@@ -0,0 +1,103 @@
+
+=== Application usage of MicroProfile Config
+
+Applications can use MicroProfile Config in the standard way to enable configuration attributes of the `ManagedExecutor` and `ThreadContext` builders to be overriden.  For example,
+
+[source, java]
+----
+@Produces @ApplicationScoped @NamedInstance("executor1")
+ManagedExecutor createExecutor(
+    @ConfigProperty(name="exec1.maxAsync", defaultValue="5") Integer a,
+    @ConfigProperty(name="exec1.maxQueued", defaultValue="20") Integer q) {
+    return ManagedExecutor.builder().maxAsync(a).maxQueued(q).build();
+}
+----
+
+MicroProfile Config can be used to override configuration attributes from the above example as follows,
+
+[source, text]
+----
+exec1.maxAsync=10
+exec1.propagated=Application,CDI
+exec1.cleared=Remaining
+----
+
+=== Container usage of MicroProfile Config
+
+The container produces an instance per unqualified `ManagedExecutor` injection point, which may optionally be annotated with the `@ManagedExecutorConfig` annotation to supply default configuration. The container also produces an instance per `ManagedExecutor` injection point that is annotated with both the `@ManagedExecutorConfig` annotation and the `@NamedInstance` qualifier. And likewise for `ThreadContext` and `@ThreadContextConfig`.
+
+In each of these cases, MicroProfile Config can be used to override the configuration of the instance that is produced by the container.
+
+To override a configuration attribute of an instance that is produced for injection into a field, specify a MicroProfile Config property with the name equal to the fully qualified class name, field name, and configuration attribute name, delimited by the `.` character.
+
+[source, java]
+----
+package org.eclipse.microprofile.example;
+...
+public class MyBean {
+    @Inject
+    ManagedExecutor exec2;
+
+    @Inject @ManagedExecutorConfig(maxAsync=3)
+    ManagedExecutor exec3;
+
+    @Inject @NamedInstance("executor4") @ManagedExecutorConfig(maxAsync=4)
+    ManagedExecutor exec4;
+
+    @Inject @NamedInstance("executor4")
+    ManagedExecutor exec5;
+}
+----
+
+[source, text]
+----
+org.eclipse.microprofile.example.MyBean.exec2.maxAsync=5
+org.eclipse.microprofile.example.MyBean.exec3.maxAsync=6
+org.eclipse.microprofile.example.MyBean.exec4.maxAsync=7
+----
+
+Note that `org.eclipse.microprofile.example.MyBean.exec4.maxAsync` overrides the configuration of the instance that is produced for injection into `exec4`. This same instance is injected into `exec5` per the matching `@NamedInstance("executor4")` qualifier.  It is an error to specify `org.eclipse.microprofile.example.MyBean.exec5.maxAsync`, which will not apply to `exec5`.
+
```

Comment:
Why is it not allowed to specify the config value? IIRC, without the annotation `@ManagedExecutorConfig()`, the maxAsync is unlimited. You can configure it via a property? When you say "it is an error", do you stop the application from starting?

Reply:
Without the ManagedExecutorConfig annotation, with only the NamedInstance Qualifier, you do not get a new instance injected.  You get the existing instance that the Qualifier matches.  If you want to use MP Config to configure that instance, this is saying that it needs to be done based on the injection point for which that instance is created, not on one of the injection point with a Qualifier that uses it.  Anticipating that users might make this mistake, it was discussed on the last MP Concurrency call that it would be helpful to detect this and consider it an error rather than ignoring it.  A decision had not been reached on how to surface the error, so I didn't elaborate on that part in the doc.  Once a decision is reached, expect this paragraph to be updated with the outcome.

Reply:
I'll assume the above answer was adequate, otherwise, we can include any further comments that you have on this paragraph under the issue where we document what the error path behavior will be, which is what it pertains to.